lib/clients.ex

```diff
@@ -6,33 +6,138 @@ defmodule GlassFactoryApi.Clients do
   alias GlassFactoryApi.Clients.Client
 
   @doc """
-  Get a client given the id
+  Return a tuple with `:ok` atom and the client from the specified ID
+  or an tuple with `:error` and a string with the error description.
 
   ## Examples
 
-      iex> GlassFactoryApi.Clients.get_client("1234")
-      %Client{
-        id: 1234,
-        name: "Google",
-        archived_at: null,
-        owner_id: 567,
-        office_id: 789
-      }
+  iex> GlassFactoryApi.Clients.get_client("1234")
+  {:ok,
+    %Client{
+      id: 1234,
+      name: "Google",
+      archived_at: null,
+      owner_id: 567,
+      office_id: 789
+    }
+  }
+
+
+  iex> GlassFactoryApi.Clients.get_client("1")
+  {:error, "Can't find a client with id 1"}
   """
   @spec get_client(integer(), map) :: {atom(), Client.t() | String.t()}
   def get_client(client_id, config \\ %{}) do
     with {:ok, %{status: 200, body: body}} <- ApiClient.get("clients/#{client_id}", config) do
       {:ok, Client.to_struct(body)}
     else
       {:ok, %{status: 404}} -> {:error, "Can't find a client with id #{client_id}"}
+      {:ok, %{body: body}} -> {:error, body}
+      {:error, error} -> {:error, error}
     end
   end
 
+  @doc """
+  Return the client from the specified ID
+  or raises an error.
+
+  ## Examples
+
+  iex> GlassFactoryApi.Clients.get_client("1234")
+  %Client{
+    id: 1234,
+    name: "Google",
+    archived_at: null,
+    owner_id: 567,
+    office_id: 789
+  }
+
+
+  iex> GlassFactoryApi.Clients.get_client("1")
+  "Can't find a client with id 1"
+  """
+  @spec get_client!(integer(), map) :: {Client.t()}
   def get_client!(client_id, config \\ %{}) do
     with {:ok, client} <- get_client(client_id, config) do
       client
     else
       {:error, error} -> raise error
     end
   end
+
+  @doc """
+  Return a tuple with `:ok` atom and all clients from an account or an tuple with `:error` and a string with the error
+  description.
+
+  ## Examples
+
+  iex> GlassFactoryApi.Clients.list_clients()
+  {:ok,
+    [
+      %Client{
+        id: 1234,
+        name: "Google",
+        archived_at: null,
+        owner_id: 567,
+        office_id: 789
+      },
+      %Client{
+        id: 1235,
+        name: "Facebook",
+        archived_at: null,
+        owner_id: 567,
+        office_id: 789
+      }
+    ]
+  }
+
+  iex> GlassFactoryApi.Clients.list_clients()
+  {:error, "econnrefused"}
+  """
+
+  @spec list_clients(map()) :: {atom(), [Client.t()] | String.t()}
+  def list_clients(config) do
+    with {:ok, %{status: 200, body: body}} <- ApiClient.get("clients", config) do
+      {:ok, Enum.map(body, &Client.to_struct(&1))}
+    else
+      {:error, error} when is_atom(error) -> {:error, Atom.to_string(error)}
```

Comment:
I think that we don't need it anymore, considering that we handle it on ApiClient 🤔

Reply:
@paulo-silva Could you give me more context about it?

Reply:
After [this PR](https://github.com/Cendrao/glass_factory_api/pull/27/files), we are handling that kind of error in `ApiClient` as you can see [here](https://github.com/Cendrao/glass_factory_api/blob/c0498aa11ad70b0dccba80bf53a907c7ff8fa025/lib/api_client.ex#L21-L26).

So you can ✂️  this guard and just handle `{:error, error}` where `error` will always be a string now.

Reply:
Thank you guys <3

Reply:
Done here: https://github.com/Cendrao/glass_factory_api/pull/25/commits/54819f4a4c6d7d8ff18d8f7a9df4a6d06a322de5